packages/leancode_hooks/lib/src/use_google_map_controller.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+/// Returns a dark-mode aware [GoogleMapController].
+Completer<GoogleMapController> useGoogleMapController({
+  required String lightStylePath,
+  required String darkStylePath,
+}) {
+  final context = useContext();
+  final brightness = MediaQuery.of(context).platformBrightness;
```

Comment:
Did you mean to use `Theme` here? Otherwise it won't react to app brightness changes and only to system wide ones

Reply:
Probably yes. In my app I tested it using system brightness only. Thanks.

Reply:
I am updating this `useGoogleMapController` hook to just get clever people to look at it. Then once it's considered "good" we can create a gist out of it and remove it from this package, so it won't be lost but also won't bloat the package.